Lib/DscResource.ps1

```diff
@@ -71,7 +71,10 @@ function GetDscResource {
     process {
         $getTargetResourceCommand = 'Get-{0}TargetResource' -f $ResourceName;
         WriteVerbose ($localized.InvokingCommand -f $getTargetResourceCommand);
-        $Parameters.Remove('Ensure');
```

Comment:
Presumably 'Ensure' still gets removed if it's not in $CommandInfo.Parameters.Keys?